### PR TITLE
Pandoc: Table edit commands

### PR DESCRIPTION
**Pull request by randomizedthinking**
The table edit function of vimwiki is quite nice. Here the code is ported to pandoc with minor modifications. It would serve as a good starting point to add more functionalities.

My initial test shows everything works:

```vim
:PandocTable       " create a table
<Tab>, <S-Tab>     " jump between cells
gqq                " format a table
...
```
Let me know how you think.

### CODE REVIEW

**Comment by fmoralesc** [comment]
Thanks! I like the idea. Some comments:

1) The `tbl.vim` file should be `table.vim`. No need to be too concise ;)
2) We should treat this as a proper module, there should be a `pandoc#table#init()` function that sets the configuration, creates the commands and sets the autocommands (check the structure of other modules, and `:help vim-pandoc-devel`). No need to touch `plugin/pandoc.vim` and `ftplugin/pandoc.vim`. Those are just for general settings. This way people can disable this functionality if they want.
3) We should have the MIT license header at the top of the file, or we should add it to the top of the repository. cc @alerque 
4) There are some TODO items in the script. Do you want to complete those before we integrate the functionality? Should we treat this as WIP?

**Comment by alerque** [comment]
Yes a license file should be placed in the top of the repository. Individual files  shouldn't bother with it in their headers.

**Comment by fmoralesc** [comment]
I agree with using the MIT license, do you too @alerque? Since I am the main contributor to the code, I think I can do it on my own, but do you think there might be trouble in doing it without contacting other contributors?

**Comment by alerque** [comment]
Yes I'm fine with MIT and it's a pretty good choice for plugins. Since you are the main contributor and the license wasn't clear before I don't think adding a permissive license like MIT is an issue. It can be a concern if implementing a more restrictive license and previous contributors can whine about that, but I don't think that is a concern here. I'd just go for it.

**Comment by fmoralesc** [comment]
Moved that to #314.

**Comment by randomizedthinking** [comment]
@fmoralesc Good that you liked the idea.

Regarding the comments you have:

- For 1. and  2., good suggestion and I can change the implementation into a module;
- For 4., I checked the two TODO items in the file, one is for improvement and  another one is a question on whether a condition check is needed for cursor on the column separator. I tested and confirmed that the current implementation works at such corner cases. So we can leave the TODO items as it is for now.

Let me know how you think.

**Comment by randomizedthinking** [comment]
Now it is converted to a module. Please review.

Per my simple tests, it works out greatly.

File: autoload/pandoc/tbl.vim [hunk 1]
```diff
@@ -0,0 +1,699 @@
+" vim:tabstop=2:shiftwidth=2:expandtab:textwidth=99
+" pandoc autoload plugin file
+" Description: Tables
+" | Easily | manageable | text  | tables | !       |
+" |--------|------------|-------|--------|---------|
+" | Have   | fun!       | Drink | tea    | Period. |
+"
+" This code is largely copied from vimwiki repository with minor modification
+" Their implementation of table editing is superb
```
[the remark this answers is not in the document]
Comment (fmoralesc) [reply autoload/pandoc/tbl.vim]: This is still there ;)

**Comment by fmoralesc** [comment]
We will probably need to add some documentation too.

File: autoload/pandoc/tbl.vim (hunk 1 above)
[the remark this answers is not in the document]
Comment (randomizedthinking) [reply autoload/pandoc/tbl.vim]: Updated now...

**Comment by randomizedthinking** [comment]
Agree, documentation is needed. I can write a simple draft when having time.

**Comment by fmoralesc** [comment]
Copying the documentation from vimwiki can be a start. We will need to add references to it and other things later.

**Comment by fmoralesc** [comment]
Ping @randomizedthinking Any update on the docs?

**Comment by randomizedthinking** [comment]
I will draft a simple doc soon... thanks for the reminder.

**Comment by randomizedthinking** [comment]
A simple write-up for the table editing is now added. Please review...

**Comment by Konfekt** [comment]
Could
```vim
inoremap <expr> <buffer> <Tab> pandoc#table#kbd_tab()
inoremap <expr> <buffer> <S-Tab> pandoc#table#kbd_shift_tab()
```
be made optional? Some may have overriden it, for example, for autocompletion. 

Perhaps one could mention (or incorporate that function) https://github.com/dhruvasagar/vim-table-mode that aligns the table automatically.

File: autoload/pandoc/table.vim [hunk 2]
```diff
@@ -0,0 +1,721 @@
...
+function! pandoc#table#Init()
+    " Table commands
+    command! -buffer -nargs=* PandocTable call pandoc#table#create(<f-args>)
+    command! -buffer PandocTableAlignQ call pandoc#table#align_or_cmd('gqq')
+    command! -buffer PandocTableAlignW call pandoc#table#align_or_cmd('gww')
+    command! -buffer PandocTableMoveColumnLeft call pandoc#table#move_column_left()
+    command! -buffer PandocTableMoveColumnRight call pandoc#table#move_column_right()
+
+    " Table mappings
+    inoremap <expr> <buffer> <Tab> pandoc#table#kbd_tab()
```
Comment (alerque) [inline autoload/pandoc/table.vim]: As @Konfekt noted in issue comments, can you please wrap these bindings in a check to make sure they are not already bound to a different key _and_ that this key isn't already bound to something else?

Comment (Konfekt) [reply autoload/pandoc/table.vim]: This [vim-script idiom](https://github.com/Konfekt/vim-smartbraces/blob/400723ea4e60665fac417be716b0189f77d78f3f/plugin/smartbraces.vim#L43) comes to mind.

**Comment by alerque** [comment]
Hmmmm, there are also some lint errors that need cleaning up.
